nox.py

```diff
@@ -94,6 +58,22 @@ def session_reqcheck(session):
         session.run('gcprepotools', command, reqfile)
 
 
+def collect_sample_dirs(start_dir, blacklist=set()):
+    """Recursively collects a list of dirs that contain tests."""
+    # Collect all the directories that have tests in them.
+    samples = []
+    for parent, subdirs, files in os.walk(start_dir):
+        if any(f for f in files if f[-8:] == '_test.py'):
+            # This dir has tests in it. Add it, and stop recursion
```

Comment:
why stop recursion? I don't think we need recursion but also don't see any reason to disable it in case someone wants multiple levels of tests to organize them better.